assets/pages/regular-contributions/components/contributionsGuestCheckoutContainer.jsx

```diff
@@ -4,48 +4,50 @@
 
 import { connect } from 'react-redux';
 import type { Dispatch } from 'redux';
-import { formIsValid } from 'helpers/checkoutForm/checkoutForm';
+import type { Contrib } from 'helpers/contributions';
+import type { UserTypeFromIdentityResponse } from 'helpers/identityApis';
+import { checkoutFormShouldSubmit, getForm } from 'helpers/checkoutForm/checkoutForm';
 import ContributionsGuestCheckout from './contributionsGuestCheckout';
 import { type State } from '../regularContributionsReducer';
 import { setStage } from '../helpers/checkoutForm/checkoutFormActions';
-import { type Action as CheckoutAction } from '../helpers/checkoutForm/checkoutFormActions';
-import { formClassName, setShouldValidateFunctions } from './formFields';
-
-
-// ----- Functions ----- //
-
-const submitYourDetailsForm = (dispatch: Dispatch<CheckoutAction>) => {
-  if (formIsValid(formClassName)) {
-    dispatch(setStage('payment'));
-    setShouldValidateFunctions.forEach(f => dispatch(f(false)));
-  } else {
-    setShouldValidateFunctions.forEach(f => dispatch(f(true)));
-  }
-};
+import { type Action as CheckoutAction, setCheckoutFormHasBeenSubmitted } from '../helpers/checkoutForm/checkoutFormActions';
+import { formClassName } from './formFields';
 
 // ----- State Maps ----- //
 
 function mapStateToProps(state: State) {
-
   return {
     amount: state.page.regularContrib.amount,
     currencyId: state.common.internationalisation.currencyId,
     country: state.common.internationalisation.countryId,
     displayName: state.page.user.displayName,
     isSignedIn: state.page.user.isSignedIn,
     stage: state.page.checkoutForm.stage,
+    userTypeFromIdentityResponse: state.page.regularContrib.userTypeFromIdentityResponse,
   };
 }
 
 const mapDispatchToProps = (dispatch: Dispatch<CheckoutAction>) => ({
   onBackClick: () => {
     dispatch(setStage('checkout'));
   },
-  onNextButtonClick:
-    () => submitYourDetailsForm(dispatch),
+  onNextButtonClick: (
+    contributionType: Contrib,
+    isSignedIn: boolean,
+    userTypeFromIdentityResponse: UserTypeFromIdentityResponse,
+  ) => {
+    if (checkoutFormShouldSubmit(
+      contributionType,
+      isSignedIn,
+      userTypeFromIdentityResponse,
+      getForm(formClassName),
+    )) {
+      dispatch(setStage('payment'));
+    }
+    dispatch(setCheckoutFormHasBeenSubmitted());
```

Comment:
probably not for this PR, but I increasingly think that this would be better called `setShouldShowFormErrors`